payload/builder/payload.go

```diff
@@ -66,16 +64,21 @@ func (pb *PayloadBuilder) RequestPayloadAsync(
 	// Assemble the payload attributes.
 	attrs, err := pb.attributesFactory.BuildPayloadAttributes(st, timestamp)
 	if err != nil {
-		return nil, common.Version{}, err
+		return nil, common.Version{}, fmt.Errorf("RequestPayloadAsync building payload attributes: %w", err)
 	}
 
-	forkVersion := pb.chainSpec.ActiveForkVersionForTimestamp(timestamp)
 	// Submit the forkchoice update to the execution client.
+	lph, err := st.GetLatestExecutionPayloadHeader()
+	if err != nil {
+		return nil, common.Version{}, fmt.Errorf("RequestPayloadAsync failed retrieving latest payload: %w", err)
+	}
+
+	forkVersion := pb.chainSpec.ActiveForkVersionForTimestamp(timestamp)
 	req := ctypes.BuildForkchoiceUpdateRequest(
 		&engineprimitives.ForkchoiceStateV1{
-			HeadBlockHash:      headEth1BlockHash,
-			SafeBlockHash:      finalEth1BlockHash,
-			FinalizedBlockHash: finalEth1BlockHash,
+			HeadBlockHash:      lph.GetBlockHash(),
+			SafeBlockHash:      lph.GetParentHash(),
+			FinalizedBlockHash: lph.GetParentHash(),
```

Comment:
this arrangement just fit, but it may change in the future:
- upon optimistic block building, blockHash it the block just verified (but not finalized), parent is already final
- upon pessimistic block building, blockHash is technically finalized but we never marked it as final (because of FinalizeBlock below)
- upon FinalizeBlock, blockHash is technically final but again we mark it as head

There is a possible future where we change this and mark blocks as final as soon as possible. If we ever do that, we could add a single extra parameter to handle the case. Until then, the changes in this PR should simplify code comprehension.